utils/record/record.go

```diff
@@ -98,7 +135,7 @@ func NewFakeEventRecorder() *FakeEventRecorder {
 			},
 			[]string{"name", "namespace", "type", "reason"},
 		),
-		nil,
+		NewFakeApiFactory(),
```

Comment:
Is there any reason to introduce this `NewFakeApiFactory()`? Thanks.

Reply:
If we donot use this we will get nil access reference error in line 207